### PR TITLE
add new test plan for deferred node pool creation

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [ ] Update changelog in CHANGELOG.md.

## Context 

Towards https://github.com/giantswarm/giantswarm/issues/14397. @calvix can you check if the test plan resembles the problem perceived in the described postmortem? 

```
awscnfm plan pl007 -h
Test plan pl007 launches a basic Tenant Cluster with a single master.
Once created a node pool is added to the cluster in order to verify
certain applications are running properly. Plan execution might take
up to 3h2m50s.

ACTION                             RETRY  WAIT
create cluster singlecontrolplane  2s     10s
verify cluster created             3m0s   30m0s
verify master ready                30s    10m0s
create nodepool defaultdataplane   2s     10s
verify worker ready                30s    10m0s
verify master hostnetworkpod       1m0s   15m0s
verify worker hostnetworkpod       1m0s   15m0s
verify kiam podandsecret           30s    10m0s
create kiam awsapicall             2s     10s
verify kiam awsapicall             5s     2m0s
delete kiam awsapicall             2s     10s
delete cluster                     2s     10s
verify cluster deleted             9m0s   1h30m0s

Usage:
  awscnfm plan pl007 [flags]

Flags:
  -h, --help                    help for pl007
  -c, --tenant-cluster string   Tenant Cluster ID to use for this particular action.
```

### CODE REVIEW

**Comment by calvix** [comment]
the order of actions is good I just wonder if we can somehow force a longer wait between master and workers as I assume  it will not cover the scenario which I encountered ( 

in my case, the cluster was created and node pool added like 30 mins later, in this cluster kiam failed since some of the installation jobs was not scheduled on worker nodes, this scenario might delay worker nodes by 5-10mins.

**Comment by xh3b4sd** [comment]
In order to delay the execution of actions I added a "cool down" property to the actions. This is now also considered for the help usage. 

```
$ awscnfm plan pl007 -h
Test plan pl007 launches a basic Tenant Cluster with a single master.
Once created a node pool is added to the cluster in order to verify
certain applications are running properly. Plan execution might take
up to 3h32m50s.

ACTION                             RETRY  WAIT     COOLDOWN
create cluster singlecontrolplane  2s     10s
verify cluster created             3m0s   30m0s    30m0s
verify master ready                30s    10m0s
create nodepool defaultdataplane   2s     10s
verify worker ready                30s    10m0s
verify master hostnetworkpod       1m0s   15m0s
verify worker hostnetworkpod       1m0s   15m0s
verify kiam podandsecret           30s    10m0s
create kiam awsapicall             2s     10s
verify kiam awsapicall             5s     2m0s
delete kiam awsapicall             2s     10s
delete cluster                     2s     10s
verify cluster deleted             9m0s   1h30m0s

Usage:
  awscnfm plan pl007 [flags]

Flags:
  -h, --help                    help for pl007
  -c, --tenant-cluster string   Tenant Cluster ID to use for this particular action.
```


```
$ awscnfm plan pl006 -h
Test plan pl006 launches a basic Tenant Cluster with a single master
and upgrades the Tenant Cluster to HA once it is up. Plan
execution might take up to 2h25m30s.

ACTION                             RETRY  WAIT
create cluster singlecontrolplane  2s     10s
verify cluster created             3m0s   30m0s
update cluster ha                  2s     10s
verify cluster ha                  1m0s   25m0s
delete cluster                     2s     10s
verify cluster deleted             9m0s   1h30m0s

Usage:
  awscnfm plan pl006 [flags]

Flags:
  -h, --help                    help for pl006
  -c, --tenant-cluster string   Tenant Cluster ID to use for this particular action.
```

File: cmd/plan/pl001/command.go [hunk 1]
```diff
@@ -46,17 +46,31 @@ func New(config Config) (*cobra.Command, error) {
...
-		d += s.Backoff.Wait()
+		d += s.Backoff.Wait() + s.CoolDown
 	}
 
 	s := "Test plan pl001 launches a basic Tenant Cluster and verifies basic criteria like\n"
 	s += "the correct number of nodes and pods are running. Plan execution might take up\n"
 	s += "to " + d.String() + ".\n\n"
 
-	t := [][]string{{"ACTION", "RETRY", "WAIT"}}
+	t := [][]string{{"ACTION", "RETRY", "WAIT", "COOLDOWN"}}
```
Comment (xh3b4sd) [inline cmd/plan/pl001/command.go:56]: The whole table magic is repeated in the commands. I would refactor this separately. I just wanted to proof this is working and works for what we want to achieve.

File: cmd/plan/pl007/plan.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// this test plan.
+var Plan = []plan.Step{
+	{
+		Action:  "create/cluster/singlecontrolplane",
+		Backoff: plan.NewBackoff(10*time.Second, 2*time.Second),
+	},
+	{
+		Action:   "verify/cluster/created",
+		Backoff:  plan.NewBackoff(30*time.Minute, 3*time.Minute),
+		CoolDown: 30 * time.Minute,
```
Comment (xh3b4sd) [inline cmd/plan/pl007/plan.go:19]: Here is the cool down defined.

**Review by njuettner — APPROVED** [review]

**Comment by paurosello** [comment]
What is the difference between cooldown and increasing the wait property?

**Comment by xh3b4sd** [comment]
Cool down is just wait without doing anything. The wait time is part of the backoff which only kicks in if an action is in fact being executed and fails. I thought about using a fake action that fails only but found this to be hacky or confusing. There was no optimal solution here.

**Comment by paurosello** [comment]
Makes sense, with this we can make "this action needs to finish in 30 min,
but wait 30 additional min after is finishes" if I understood correctly

**Comment by xh3b4sd** [comment]
Exactly like that.

**Review by calvix — APPROVED** [review]
nice

**Pull request merged**